core/trino-main/src/main/java/io/trino/eventlistener/EventListenerManager.java

```diff
@@ -134,38 +143,65 @@ private static Map<String, String> loadEventListenerProperties(File configFile)
 
     public void queryCompleted(Function<Boolean, QueryCompletedEvent> queryCompletedEventProvider)
     {
-        for (EventListener listener : configuredEventListeners.get()) {
-            QueryCompletedEvent event = queryCompletedEventProvider.apply(listener.requiresAnonymizedPlan());
-            try {
-                listener.queryCompleted(event);
-            }
-            catch (Throwable e) {
-                log.warn(e, "Failed to publish QueryCompletedEvent for query %s", event.getMetadata().getQueryId());
+        try (TimeStat.BlockTimer ignored = queryCompletedTime.time()) {
+            for (EventListener listener : configuredEventListeners.get()) {
```

Comment:
you have `doQueryCompleted` method. That way you would avoid changing existing code:

```
    public void queryCompleted(Function<Boolean, QueryCompletedEvent> queryCompletedEventProvider)
    {
         try (TimeStat.BlockTimer ignored = queryCompletedTime.time()) {
               doQueryCompleted(queryCompletedEventProvider);
         }
   }

    private void doQueryCompleted(Function<Boolean, QueryCompletedEvent> queryCompletedEventProvider)
    {
        for (EventListener listener : configuredEventListeners.get()) {
            QueryCompletedEvent event = queryCompletedEventProvider.apply(listener.requiresAnonymizedPlan());
            try {
                listener.queryCompleted(event);
            }
            catch (Throwable e) {
                log.warn(e, "Failed to publish QueryCompletedEvent for query %s", event.getMetadata().getQueryId());
....
```

Reply:
Done